session.go

```diff
@@ -1611,12 +1611,25 @@ func (idxs indexSlice) Swap(i, j int)      { idxs[i], idxs[j] = idxs[j], idxs[i]
 
 func simpleIndexKey(realKey bson.D) (key []string) {
 	for i := range realKey {
+		var vi int
 		field := realKey[i].Name
-		vi, ok := realKey[i].Value.(int)
-		if !ok {
+
```

Comment:
I haven't checked the call site, but is downcasting int64 to in  really necessary ? Mongo itself does handle 64bit integers IIRC.

Reply:
After looking at the code properly, the only valid values are "1" and "-1" so casting isn't an issue.